addons/mail/static/tests/thread/unread_messages_banner.test.js

```diff
@@ -220,3 +222,30 @@ test("sidebar and banner counters display same value", async () => {
         parent: [".o-mail-DiscussSidebarChannel", { text: "Bob" }],
     });
 });
+
+test("mobile: mark as read when opening chat", async () => {
+    const pyEnv = await startServer();
+    const bobPartnerId = pyEnv["res.partner"].create({ name: "bob" });
+    const channelId = pyEnv["discuss.channel"].create({
+        channel_type: "chat",
+        channel_member_ids: [
+            Command.create({ partner_id: serverState.partnerId }),
+            Command.create({ partner_id: bobPartnerId }),
+        ],
+    });
+    pyEnv["mail.message"].create({
+        body: "Hello!",
+        model: "discuss.channel",
+        author_id: bobPartnerId,
+        res_id: channelId,
+    });
+    patchUiSize({ size: SIZES.SM });
+    await start();
+    await openDiscuss();
+    await contains("button.active", { text: "Inbox" });
+    await click("button", { text: "Chat" });
+    await click(".o-mail-NotificationItem", { text: "bob" });
+    await click(".o-mail-ChatWindow-command[title*='Close Chat Window']");
+    await click(".o-mail-NotificationItem", { text: "bob" });
+    await contains(".o-mail-Thread-banner", { count: 0 });
```

Comment:
Negative assertion without equivalent positive makes test poor: assertion can pass when it shouldn't (e.g. classnames have changed or removed or replaced).

I would assert presence of badge in chat list before, and then it's removed.